include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -748,6 +749,8 @@ struct ConstSessionImpl : Base<T> {
   TypeInfo GetOverridableInitializerTypeInfo(size_t index) const;  ///< Wraps OrtApi::SessionGetOverridableInitializerTypeInfo
 };
 
+using RunAsyncCallbackStdFn = std::function<void(std::vector<Value>&, Status)>;
```

Comment:
Is taking a ref to the vector correct here?

Reply:
which option makes it more sound?

Reply:
maybe `const std::vector<Value> &`?

Please first think about [the other comments](https://github.com/microsoft/onnxruntime/pull/16613/files/42837092dba341c20515728e73ce1b678d80f58f..0ac824e6cfaf40a2693710078d8f3c7ae90a4cce#r1258916069) mentioned in C-API and if possible we can remove this from the callback signature.

Reply:
what prevent customer from operate on the vector freely?

Reply:
this vector is a parameter of a callback function. users are expected to read but not modify it.

Reply:
The code has been dropped.